Configuration.mk

```diff
@@ -286,28 +277,57 @@ override WLFLAGS_rv32imac += $(WLFLAGS_rv32)
 
 # Set the system libraries we link against for RISC-V. We support C++ apps by
 # default.
-override LINK_LIBS_rv32 += \
-      -lgcc -lstdc++ -lsupc++
+ifeq ($(PICOLIBC),)
+      override LINK_LIBS_rv32 += \
+            -lgcc -lstdc++ -lsupc++
 
-override LINK_LIBS_rv32i    += $(LINK_LIBS_rv32)
-override LINK_LIBS_rv32imc  += $(LINK_LIBS_rv32)
-override LINK_LIBS_rv32imac += $(LINK_LIBS_rv32)
+      # Use precompiled libaries we provide to link against.
+      override LEGACY_LIBS_rv32i += \
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32i/libc.a\
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32i/libm.a
+
+      override LEGACY_LIBS_rv32im += \
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32im/libc.a\
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32im/libm.a
+
+      override LEGACY_LIBS_rv32imc += $(LEGACY_LIBS_rv32im)
+
+      override LEGACY_LIBS_rv32imac += \
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32imac/libc.a\
+            $(TOCK_USERLAND_BASE_DIR)/newlib/rv32/rv32imac/libm.a
+else
+      override LEGACY_LIBS_rv32i += \
+            /usr/lib/picolibc/riscv64-unknown-elf/lib/rv32i/ilp32/libc.a\
```

Comment:
We can't assume these paths can we?

If this needs to be customized at this level then I don't think we are ready to assume RISC-V support yet.

Reply:
They are pretty standard paths, so I would say we could. But I have removed them anyway